docs/proposals/application/single_cluster_deployment_application.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Single cluster deployment application
+authors:
+- @yeyeye2333
+reviewers:
+approvers:
+
+creation-date: 2024-07-24
+
+---
+
+## Single Cluster Deployment Application
+
+### Summary
+
+This proposal aims to modify the application API to allow setting the application's destination to a cluster, reducing the operational steps required for Kurator in a single-cluster environment and increasing Kurator's flexibility.
+
+### Motivation
+
+Kurator leverages Fleet for multi-cluster management. However, this has resulted in the current application deployment being tightly coupled with Fleet, even requiring Fleet creation for single-cluster usage. Therefore, we aim to decouple the single-cluster client from Fleet.
+
+#### Goals
+
+- Modify the application API to support setting the application's destination to a cluster.
+- Simplify the deployment process for single-cluster environments by eliminating the need for Fleet.
+
+### Proposal
+
+Currently, the application registration is handled in the fleet-manager. The proposal is to move the application registration to the cluster-operator to support setting the application's destination to a cluster.
+
+### Design Details
+
+We will delve into the API design required to support setting the application's destination to a cluster. The following are the specific changes to the application:
```

Comment:
Can you describe on how deploying an application in a single cluster through kurator should work if we follow your changes?

Reply:
like this application.yaml

```yaml
apiVersion: apps.kurator.dev/v1alpha1
kind: Application
metadata:
	...
spec:
	...
  syncPolicies:
    - destination:
        cluster: 
          name: quickstart
```

Reply:
Can it be designed so that it is the current cluster deployment when no destination is specified?

Reply:
Does the current cluster refer to deploying directly to the host cluster when there is no member cluster?

Reply:
Yes.

Reply:
Do we still need to provide the above API to specify a cluster?

Reply:
If you're deploying to a single cluster, you don't need this API to specify the cluster.